File: app/src/main/java/org/deiverbum/app/model/BibleBooks.java
package org.deiverbum.app.model;

import android.text.SpannableStringBuilder;

import org.deiverbum.app.utils.Utils;

public class BibleBooks {
    private int id;
    private String name;
    private String description;
    private String intro;

    public BibleBooks() {
    }

    public BibleBooks(int id, String name, String description) {
        this.name = name;
        this.id = id;
        this.description = description;

    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getIntro() {
        return intro;
    }

    public void setIntro(String intro) {
        this.intro = intro;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }


    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public SpannableStringBuilder getForView() {
        SpannableStringBuilder sb=new SpannableStringBuilder();
        sb.append(Utils.fromHtml(getIntro()));
        return sb;
    }

    public StringBuilder getForRead() {
        StringBuilder sb=new StringBuilder();
        sb.append(Utils.fromHtml(Utils.stripQuotation(getIntro())));
        return sb;
    }
}

File: app/src/main/java/org/deiverbum/app/model/BibleBooks.kt
package org.deiverbum.app.model

import android.text.SpannableStringBuilder
import org.deiverbum.app.utils.ColorUtils
import org.deiverbum.app.utils.Utils

class BibleBooks(var id: Int, var name: String?, var description: String?) {
    var intro: String? = null

    fun getForView(isNightMode: Boolean): SpannableStringBuilder {
        ColorUtils.isNightMode = isNightMode
        val sb = SpannableStringBuilder()
        sb.append(Utils.fromHtml(intro))
        return sb
    }

    val forRead: StringBuilder
        get() {
            val sb = StringBuilder()
            sb.append(
                Utils.fromHtml(
                    Utils.stripQuotation(
                        intro
                    )
                )
            )
            return sb
        }
}